sdformat_mjcf/src/sdformat_mjcf/mjcf_to_sdformat/converters/world.py

```diff
@@ -77,152 +137,45 @@ def mjcf_worldbody_to_sdf(mjcf_root, physics, world,
-                                                  body_name,
-                                                  child_name_result)
-                    serial_joints_created.append(child_name_result)
-                    serial_sdf_joints_created.append(joint_sdf)
-                    body_name = child_name_result
-
-                if joint_sdf is not None:
-                    model.add_joint(joint_sdf)
-            if len(body.joint) == 0 and body.freejoint is None:
-                joint_sdf = add_fixed_joint(body_parent_name, body.name)
-                model.add_joint(joint_sdf)
-
-            # We might have more than one link, if there is the case,
-            # we should remove visuals and colllisions in all the previous
-            # link only if the joint is in the same place
-            if len(serial_link_created) > 1:
-                if serial_sdf_joints_created[0] is not None and \
-                   serial_sdf_joints_created[-1] is not None:
-                    if serial_sdf_joints_created[0].raw_pose() == \
-                       serial_sdf_joints_created[-1].raw_pose():
-                        serial_link_created[-1].add_visual(
-                            link.visual_by_index(0))
-                        serial_link_created[-1].add_collision(
-                            link.collision_by_index(0))
-                        serial_link_created[-1].set_inertial(link.inertial())
-                        link.clear_visuals()
-                        link.clear_collisions()
-                        link.inertial().mass_matrix().set_mass(0)
-
-            # Add all links to the world
-            for i in range(len(serial_link_created)):
-                model.add_link(serial_link_created[i])
-
-            model.add_link(link)
-            iterate_bodies(body.body,
-                           model,
-                           body_name)
-    iterate_bodies(body, model)
+    models = []
+    root_body_to_model = {}
+    for body in mjcf_root.worldbody.body:
+        model = sdf.Model()
+        link = _add_body_to_model(body, physics, model, modifiers)
+        model.set_name(f"model_for_{link.name()}")
+        models.append(model)
+        root_body_to_model[body] = model
+
+    # Walk up the hierarchy until we find body in root_body_to_model since
+    # we only store top level bodies (kinematic roots) in root_body_to_model
+    def _get_root_body(body):
+        while body.parent != body.root.worldbody:
+            body = body.parent
+        return body
+
+    def _get_model_from_sensor(sensor):
+        if hasattr(sensor, "site"):
+            return root_body_to_model[_get_root_body(sensor.site.parent)]
+
+        raise RuntimeError(
+            f"Sensor {sensor} not attached to a site or body")
 
     if mjcf_root.sensor is not None:
         for accel in mjcf_root.sensor.accelerometer:
             if accel is not None:
+                model = _get_model_from_sensor(accel)
```

Comment:
`model` might be `None`, this is not checked in `mjcf_*_sensor_to_sdf`. Should we add some logic here ?

Reply:
Unless I'm mistaken, if `model` is None, then something is very wrong, so I'd rather have it throw an exception.